RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,258 @@
+---
+RFC: RFCXXX-Native-Json-Adapters.md
+Author: Jim Truher
+Status: Draft
+SupercededBy: 
+Version: 0.8
+Area: engine
+Comments Due: March 1, 2023
+Plan to implement: Yes
+---
+
+# Title
+
+Improving the User Experience of Native Applications
+
+## Motivation
+
+As a user I can interact with native applications which can produce a more object like experience.
+
+### Native Applications which Emit Structured Data
+
+Native command execution usually results in textual output which is a fairly poor experience for our users.
+Some of our users use tools like `Crescendo` to allow those tools to "play nice" in the PowerShell arena,
+but the problem of output handling is still troublesome.
+
+Some applications (docker, kubectl, others) can emit in `JSON` which is much easier for our users to convert to useful results in PowerShell,
+but the user still has to know the incantation to have the tool emit `JSON` and then they must pipe that output to `Convert-FromJson`.
+It would be a much better experience if the user could simply type:
+
+```powershell
+PS> docker image ls
+```
+
+rather than
+
+```powershell
+PS> docker image ls --format='{{json .}}' | ConvertFrom-Json | Format-Table
+```
+
+There are two aspects of making this work well:
+
+- Automatically adding `ConvertFrom-Json` to the pipeline following the native command to automatically convert `Json` to object
```

Comment:
How are we going to to know we need to automatically add the `ConvertFrom-Json` to the pipeline or not? Cause we dont need an adapter for example on line 37 to work

Reply:
Think may have answered on line 80 but does that mean still requiring a basic adapter to only call `ConvertFrom-Json` to know if it is used or not

Reply:
Maybe clarify here its always the adapter and not ConvertFrom-JSON being put at the end